internal/crd/crd.go

```diff
@@ -63,6 +64,38 @@ func Validate(ctx context.Context, cl client.Client, newCrd *apiextensionsv1.Cus
 	return nil
 }
 
+// CreateOrUpdateCRD performs all the necessary actions to either Create or Update a CRD,
+// including running validation when an Update is required.
+func CreateOrUpdateCRD(ctx context.Context, cl client.Client, newCrd *apiextensionsv1.CustomResourceDefinition) error {
+	// First try to create the CRD
+	err := cl.Create(ctx, newCrd)
+	if err != nil && apierrors.IsAlreadyExists(err) {
+		// If it exists already check if it's safe to update
+		err = Validate(ctx, cl, newCrd)
+		if err != nil {
+			// If not, return reason why we cannot update safely
+			return err
+		}
```

Comment:
I think we should drop this `Validate` check. IIRC, we use this in the CRD validator webhook, and I'm pretty sure our plan is still to get rid of that, or at least turn it off.